node.tf

```diff
@@ -103,21 +37,23 @@ resource "aws_launch_configuration" "nodes_green" {
   }
 }
 
-resource "aws_autoscaling_group" "nodes_green" {
-  desired_capacity      = var.nodes_green_desired_capacity
-  launch_configuration  = aws_launch_configuration.nodes_green.id
-  max_size              = var.nodes_green_max_size
-  min_size              = var.nodes_green_min_size
-  name                  = "${var.environment_name}-nodes-green"
-  max_instance_lifetime = var.nodes_green_max_instance_lifetime
+resource "aws_autoscaling_group" "asg_nodes" {
+  for_each = var.asg_nodes
+
+  desired_capacity      = each.value.nodes_desired_capacity
+  launch_configuration  = aws_launch_configuration.asg_nodes[each.key].id
+  max_size              = each.value.nodes_max_size
+  min_size              = each.value.nodes_min_size
+  name                  = "${var.environment_name}-nodes-${each.key}"
+  max_instance_lifetime = each.value.max_instance_lifetime
 
-  vpc_zone_identifier = length(var.nodes_green_subnet_ids) == 0 ? (var.nodes_in_public_subnet ? aws_subnet.public.*.id : aws_subnet.private.*.id) : var.nodes_green_subnet_ids
+  vpc_zone_identifier = length(each.value.subnet_ids) == 0 ? (each.value.nodes_in_public_subnet ? aws_subnet.public.*.id : aws_subnet.private.*.id) : each.value.subnet_ids
 
-  enabled_metrics = var.enabled_metrics_asg
+  enabled_metrics = each.value.node_enabled_metrics
```

Comment:
Do a lookup. If it is empty then set to the value https://github.com/opszero/terraform-aws-kubespot/pull/333/files#diff-05b5a57c136b6ff596500bcbfdcff145ef6cddea2a0e86d184d9daa9a65a288eR318